misc/style/run-clang-tidy.py

```diff
@@ -83,12 +98,17 @@ def check_search_code_with_clang_tidy():
         "readability-static-definition-in-anonymous-namespace",
         "readability-uniqueptr-delete-release",
         ]
+    config = """{CheckOptions: [\
+        {key: performance-unnecessary-value-param.AllowedTypes, value: "StateID"},\
```

Comment:
This exception is a bit hard to find in the middle of the file. Maybe make this a global variable (`LIGHTWEIGHT_TYPES = ["StateID"]` or something) at the start of the file next to `IGNORES`?